Library/Homebrew/cmd/update.rb

```diff
@@ -319,18 +319,51 @@ def empty?
   def dump
     # Key Legend: Added (A), Copied (C), Deleted (D), Modified (M), Renamed (R)
 
+    update_renamed
+
     dump_formula_report :A, "New Formulae"
     dump_formula_report :M, "Updated Formulae"
+    dump_formula_report :R, "Renamed Formulae"
     dump_formula_report :D, "Deleted Formulae"
   end
 
-  def select_formula(key)
-    fetch(key, []).map do |path|
+  def update_renamed
+    @hash[:R] = []
+
+    fetch(:D, []).each do |path|
       case path.to_s
       when HOMEBREW_TAP_PATH_REGEX
-        "#{$1}/#{$2.sub("homebrew-", "")}/#{path.basename(".rb")}"
+        user = $1
+        repo = $2.sub("homebrew-", "")
+        oldname = path.basename(".rb").to_s
+        next unless newname = Tap.new(user, repo).formula_renames[oldname]
+      else
+        oldname = path.basename(".rb").to_s
+        next unless newname = FORMULA_RENAMES[oldname]
+      end
+
+      if fetch(:A, []).include?(newpath = path.dirname.join("#{newname}.rb"))
+        @hash[:R] << [path, newpath]
+      end
+    end
+
+    @hash[:A] -= @hash[:R].map(&:last) if @hash[:A]
+    @hash[:D] -= @hash[:R].map(&:first) if @hash[:D]
+  end
+
+  def select_formula(key)
+    fetch(key, []).map do |path, newpath|
+      if path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+        tap = "#{$1}/#{$2.sub("homebrew-", "")}"
+        if newpath
+          "#{tap}/#{path.basename(".rb")} -> #{tap}/#{newpath.basename(".rb")}"
```

Comment:
I found it's strange to have a method named `select_formula` returning string which is not a formula name. I suggest you create a new method to do this.